dvc/schema.py

```diff
@@ -0,0 +1,41 @@
+from dvc.stage.params import StageParams
+from dvc import dependency, output
+
+from voluptuous import Any, Schema, Optional
+
+
+SCHEMA = {
+    StageParams.PARAM_MD5: output.CHECKSUM_SCHEMA,
+    StageParams.PARAM_CMD: Any(str, None),
+    StageParams.PARAM_WDIR: Any(str, None),
+    StageParams.PARAM_DEPS: Any([dependency.SCHEMA], None),
+    StageParams.PARAM_OUTS: Any([output.SCHEMA], None),
+    StageParams.PARAM_LOCKED: bool,
+    StageParams.PARAM_META: object,
+    StageParams.PARAM_ALWAYS_CHANGED: bool,
+}
+
+LOCK_FILE_STAGE_SCHEMA = {
+    StageParams.PARAM_CMD: str,
+    StageParams.PARAM_DEPS: {str: output.CHECKSUM_SCHEMA},
+    StageParams.PARAM_OUTS: {str: output.CHECKSUM_SCHEMA},
+}
+LOCKFILE_SCHEMA = {str: LOCK_FILE_STAGE_SCHEMA}
+
+SINGLE_PIPELINE_STAGE_SCHEMA = {
+    str: {
+        StageParams.PARAM_CMD: str,
+        Optional(StageParams.PARAM_WDIR): str,
+        Optional(StageParams.PARAM_DEPS): [str],
+        Optional(StageParams.PARAM_OUTS): [str],
+        Optional(StageParams.PARAM_LOCKED): bool,
+        Optional(StageParams.PARAM_META): object,
+        Optional(StageParams.PARAM_ALWAYS_CHANGED): bool,
+        Optional("persist"): [str],
```

Comment:
I totally forgot about `persist`. 😂

Reply:
It'd be great if we don't even dump `cache: True, persist: False`, etc. on outs for output pointer file.

Reply:
@skshetry Totally agree with that. We also don't need to dump default values. That is also something that we've been talking about for a long time, just didn't get to it :slightly_smiling_face: Maybe leave it for later PR, if it doesn't matter for your PR.